trie/secure_trie.go

```diff
@@ -270,7 +275,7 @@ func (t *StateTrie) Copy() *StateTrie {
 	return &StateTrie{
 		trie:        *t.trie.Copy(),
 		db:          t.db,
-		secKeyCache: t.secKeyCache,
+		secKeyCache: maps.Clone(t.secKeyCache),
```

Comment:
Maybe sharing the `secKeyCache` to a new `StateTrie` instance is not right. Clone it more in line with the functional requirements.